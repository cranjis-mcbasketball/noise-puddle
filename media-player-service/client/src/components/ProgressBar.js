import React from 'react';
import moment from 'moment';

// styles
import ProgressBarStyles from '../styled-components/ProgressBar.js';
import CurrentTimeStyle from '../styled-components/CurrentTimeStyle.js';
import DurationStyle from '../styled-components/DurationStyle.js';

class ProgressBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      height: []
    }
  }

  componentDidMount() {
    const bars = [];
    for (let i = 0; i < 283; i++) {
      let randomBars = Math.random() * 100
      this.state.height.push(randomBars)
    }
  }

  render() {
    const progressPercent = this.props.currentTime / this.props.duration;
    const containerWidth = 850;
    const progressWidth = containerWidth * progressPercent;

    const currentSongDuration = this.props.duration;
    const songDurationFormatted = moment.utc(currentSongDuration * 1000).format('mm:ss');

    const currentSongTime = this.props.currentTime;
    const currentSongTimeFormatted = moment.utc(currentSongTime * 1000).format('mm:ss');

    const bars = [];
    const progress = [];
    for (let i = 0; i < 283; i++) {
      bars.push(
        <div style={{
          display: 'inline-block',
          width: '2px',
          height: this.state.height[i + 1] + 'px',
          background: 'white',
          marginRight: '1px',
          position: 'relative',
        }}></div>
      )
      progress.push(
        <div style={{
          display: 'inline-block',
          height: this.state.height[i + 1] + 'px',
          width: '2px',
          backgroundImage: 'linear-gradient(-60deg, #ff5858 0%, #f09819 100%)',
          position: 'relative',
          marginRight: '1px',
          marginBottom: '-83.5px',
          zIndex: '1',
        }}></div>
      )
    }

    const difference = 0 + progressWidth
    return (
      <ProgressBarStyles>
        <CurrentTimeStyle>
          {currentSongTimeFormatted}
        </CurrentTimeStyle>
        <div className="progress" style={{
          width: difference + 'px',
          position: 'absolute',
          background: 'none',
          height: '100px',
          overflow: 'hidden',
          whiteSpace: 'nowrap',
        }}>
          {progress}
        </div>
        <div style={{
          position: 'absolute',
        }}>
          {bars}
        </div>
        <DurationStyle>
          {songDurationFormatted}
        </DurationStyle>
      </ProgressBarStyles>
    )
  }
}

export default ProgressBar;