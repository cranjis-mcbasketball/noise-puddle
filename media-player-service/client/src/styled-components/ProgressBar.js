import styled from 'styled-components';

export default styled.div`
  background: none;
  height: 120px;
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 370px;
`