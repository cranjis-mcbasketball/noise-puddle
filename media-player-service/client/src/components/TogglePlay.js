import React from 'react';
import ProgressBar from './ProgressBar.js';

import ButtonStyle from '../styled-components/Play-Pause.js';

class TogglePlay extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isPlaying: false,
      isLoaded: false
    }
    this.toggle = this.toggle.bind(this);
    this.updateTime = this.updateTime.bind(this);
  }

  toggle() {
    if (this.state.isLoaded) {
      const myAudio = document.getElementById('myAudio');
      myAudio.volume = 0.05;
      this.setState({ isPlaying: !this.state.isPlaying }, () => {
        this.state.isPlaying ? myAudio.play() : myAudio.pause();
      });
    }
  }

  componentDidMount() {
    const myAudio = document.getElementById('myAudio');
    myAudio.onloadedmetadata = (e) => this.updateTime(e);
    myAudio.ontimeupdate = (e) => this.updateTime(e);
    myAudio.oncanplay = () => this.setState({ isLoaded: true });
  }

  updateTime(e) {
    this.props.setTime(e.target.currentTime);
    this.props.setDuration(e.target.duration);
  }

  renderProgressBar() {
    <ProgressBar
      currentTime={this.state.currentTime}
      duration={this.state.duration} />
  }

  render() {
    const playImg = 'https://sdc-media-player.s3.us-east-2.amazonaws.com/buttons/playButton.png';
    const pauseImg = 'https://sdc-media-player.s3.us-east-2.amazonaws.com/buttons/pauseButton.png';
    return (
      <span>
        <audio
          id="myAudio"
          src={this.props.currentSong.song_url}
          preload="auto">
        </audio>
        <a onClick={this.toggle}>
          {this.state.isPlaying ?
            <ButtonStyle
              src={require = (pauseImg)} /> :
            <ButtonStyle
              src={require = (playImg)} />}
        </a>
        {this.renderProgressBar()}
      </span>
    )
  }
}

export default TogglePlay;