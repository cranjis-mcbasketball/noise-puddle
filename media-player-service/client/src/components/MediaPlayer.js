import React, { useState } from 'react';
import moment from 'moment';
import TogglePlay from './TogglePlay.js';
import ProgressBar from './ProgressBar.js';

// styles
import Container from '../styled-components/Container.js';
import Image from '../styled-components/Image.js';
import Date_Genre_Style from '../styled-components/Date-Genre.js';
import Band from '../styled-components/BandName.js';
import SongStyle from '../styled-components/SongName.js';
import NameContainer from '../styled-components/SongBandContainer.js';

const MediaPlayer = ({ currentSong }) => {
  const now = moment();
  let bruh;
  if (currentSong.release_date) {
    bruh = currentSong.release_date.split('-').join(' ');
  }
  const created = moment(bruh);
  const timeElapsed = created.from(now);

  const [time, setTime] = useState(0);
  const [duration, setDuration] = useState(0);

  return (
    <Container>
      <Image src={`${currentSong.album_image}.jpg`} />
      <Date_Genre_Style>
        {timeElapsed}<br />
        {currentSong.music_genre}
      </Date_Genre_Style>
      <NameContainer>
        <TogglePlay
          currentSong={currentSong}
          setTime={setTime}
          setDuration={setDuration}
        />
        <Band>{currentSong.band_name}</Band>
        <SongStyle>{currentSong.song_name}</SongStyle>
      </NameContainer>
      <ProgressBar
        currentTime={time}
        duration={duration}
      />
    </Container>
  )
}

export default MediaPlayer;