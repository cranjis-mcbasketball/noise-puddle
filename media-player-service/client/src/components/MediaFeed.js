import React from 'react';
import ReactDOM from 'react-dom'
import axios from 'axios';
import MediaPlayer from './MediaPlayer.js';

class MediaFeed extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentSong: []
    }
  }

  componentDidMount() {
    axios.get('http://localhost:3305/songs/9999999')
      .then(music => {
        const song = {
          song_name: music.data.title,
          music_genre: music.data.genre,
          band_name: music.data.name,
          album_image: music.data.art,
          song_url: music.data.url,
          release_date: music.data.release
        }
        this.setState({
          currentSong: song
        })
      })
      .catch(function (error) {
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        <div>
          <MediaPlayer
            currentSong={this.state.currentSong}
          />
        </div>
      </div>
    )
  }
}

export default MediaFeed;