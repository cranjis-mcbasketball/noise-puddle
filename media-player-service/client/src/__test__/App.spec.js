import React from 'react';
import { shallow, mount } from 'enzyme';
import regeneratorRuntime from "regenerator-runtime";

import MediaFeed from '../components/MediaFeed.js';
import ProgressBar from '../components/ProgressBar.js';
import TogglePlay from '../components/TogglePlay.js';
import MediaPlayer from '../components/MediaPlayer.js';

describe('My Test Suite', () => {
  it('My Test Case', () => {
    expect(true).toEqual(true);
  });
});

describe('MediaFeed', () => {
  test('should render snapshot', () => {
    const wrapper = shallow(<MediaFeed />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should exist', () => {
    const wrapper = shallow(<MediaFeed />);
    expect(wrapper.exists()).toBe(true);
  });
  test('should not have class names', () => {
    const wrapper = shallow(<MediaFeed />);
    expect(wrapper.hasClass()).toBe(false);
  });
});

describe('Play/Pause button', () => {
  const mockFn = jest.fn();
  test('play button should be defined', () => {
    expect(TogglePlay).toBeDefined();
  });
});

describe('ProgressBar', () => {
  test('should render snapshot', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should exist', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(wrapper.exists()).toBe(true);
  });
  test('should not have class names', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(wrapper.hasClass()).toBe(false);
  });
})