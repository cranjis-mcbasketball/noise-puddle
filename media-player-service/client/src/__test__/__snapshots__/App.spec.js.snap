// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaFeed should render snapshot 1`] = `ShallowWrapper {}`;

exports[`ProgressBar should render snapshot 1`] = `ShallowWrapper {}`;
